keras/utils/io_utils.py

```diff
@@ -155,3 +164,185 @@ def ask_to_proceed_with_overwrite(filepath):
         return False
     print('[TIP] Next time specify overwrite=True!')
     return True
+
+
+class H5Dict(object):
+    """ A dict-like wrapper around h5py groups (or dicts).
+
+    This allows us to have a single serialization logic
+    for both pickling and saving to disk.
+
+    Note: This is not intended to be a generic wrapper.
+    There are lot of edge cases which have been hardcoded,
+    and makes sense only in the context of model serialization/
+    deserialization.
+    """
+
+    def __init__(self, path, mode='a'):
+        if isinstance(path, h5py.Group):
+            self.data = path
+            self._is_file = False
+        elif isinstance(path, str):
+            self.data = h5py.File(path,)
+            self._is_file = True
+        elif isinstance(path, dict):
+            self.data = path
+            self._is_file = False
+            self.data['_is_group'] = True
```

Comment:
Not sure that I get what this flag is, maybe add some quick doc?

Reply:
Whether a dict is user data (`f[attr] = {'x' : 'y'}`)  or a group (`new_group = f['new_group']`).